### ISSUE TITLE
Memoize nested getter?

### ISSUE THREAD

**Issue by Yomguithereal** [Yomguithereal/baobab#200]

**Comment by AutoSponge** [comment]
I have a lib I wrote a while ago that memoizes the accessor of an arbitrary depth (but doesn't memoize the path). It's been very stable. I could take a swing at implementing the same thing in baobab if you're interested.

https://www.npmjs.com/package/perfget

**Comment by Yomguithereal** [comment]
This is interesting indeed @AutoSponge. Let me check that. I was thinking of memoizing some things lately to boost getter perfs and this is really walk along this purpose.

My first guess was to keep the gotten reference and only walk the path once on getters and refresh on cursor update. I should read the documentation of your library to check what you propose.

In the meantime, if you want to check what is currently done when retrieving nested path, you can check this [code](https://github.com/Yomguithereal/baobab/blob/master/src/helpers.js#L196-L237).

**Comment by Yomguithereal** [comment]
By the way, do you have any benchmark somewhere @AutoSponge?

**Comment by AutoSponge** [comment]
I made this: https://github.com/AutoSponge/bench

The readme results are from Node, but seem consistent with Chrome. Your mileage may vary with different environments. 

While I haven't proven it, it's possible that the dynamically constructed 'getter' can't be JIT compiled like the adaptive algorithm. Also, once proper tail calls exist, the recursive and adaptive should have the same results.

Overall, perfget's approach (I altered it slightly to have the same signature as other methods) has the best (or tied for best) performance when considering multiple environments and varying depth.
